From patchwork Mon Sep  3 13:35:49 2018
Content-Type: text/plain; charset="utf-8"
MIME-Version: 1.0
Content-Transfer-Encoding: 7bit
X-Patchwork-Submitter: Wei Yang <richard.weiyang@gmail.com>
X-Patchwork-Id: 10585857
Return-Path: <kvm-owner@kernel.org>
Received: from mail.wl.linuxfoundation.org (pdx-wl-mail.web.codeaurora.org
 [172.30.200.125])
	by pdx-korg-patchwork-2.web.codeaurora.org (Postfix) with ESMTP id E11C413BB
	for <patchwork-kvm@patchwork.kernel.org>;
 Mon,  3 Sep 2018 13:36:01 +0000 (UTC)
Received: from mail.wl.linuxfoundation.org (localhost [127.0.0.1])
	by mail.wl.linuxfoundation.org (Postfix) with ESMTP id D306B2945C
	for <patchwork-kvm@patchwork.kernel.org>;
 Mon,  3 Sep 2018 13:36:01 +0000 (UTC)
Received: by mail.wl.linuxfoundation.org (Postfix, from userid 486)
	id C6E772947A; Mon,  3 Sep 2018 13:36:01 +0000 (UTC)
X-Spam-Checker-Version: SpamAssassin 3.3.1 (2010-03-16) on
	pdx-wl-mail.web.codeaurora.org
X-Spam-Level: 
X-Spam-Status: No, score=-8.0 required=2.0 tests=BAYES_00,DKIM_SIGNED,
	DKIM_VALID,DKIM_VALID_AU,FREEMAIL_FROM,MAILING_LIST_MULTI,RCVD_IN_DNSWL_HI
	autolearn=ham version=3.3.1
Received: from vger.kernel.org (vger.kernel.org [209.132.180.67])
	by mail.wl.linuxfoundation.org (Postfix) with ESMTP id 3A6B92946B
	for <patchwork-kvm@patchwork.kernel.org>;
 Mon,  3 Sep 2018 13:36:00 +0000 (UTC)
Received: (majordomo@vger.kernel.org) by vger.kernel.org via listexpand
        id S1727380AbeICR4K (ORCPT
        <rfc822;patchwork-kvm@patchwork.kernel.org>);
        Mon, 3 Sep 2018 13:56:10 -0400
Received: from mail-pf1-f196.google.com ([209.85.210.196]:34693 "EHLO
        mail-pf1-f196.google.com" rhost-flags-OK-OK-OK-OK) by vger.kernel.org
        with ESMTP id S1727112AbeICR4K (ORCPT <rfc822;kvm@vger.kernel.org>);
        Mon, 3 Sep 2018 13:56:10 -0400
Received: by mail-pf1-f196.google.com with SMTP id k19-v6so251684pfi.1
        for <kvm@vger.kernel.org>; Mon, 03 Sep 2018 06:35:58 -0700 (PDT)
DKIM-Signature: v=1; a=rsa-sha256; c=relaxed/relaxed;
        d=gmail.com; s=20161025;
        h=from:to:cc:subject:date:message-id;
        bh=4xx/AtucN5CvwisH9ffjv+eJjFMQ1dh59OgKza6nQas=;
        b=R8Yn3cX6008dLmSVEkdHmGeE01I1zWogApUmHpO1hXGeWmbpcr30gNNcJs1kCMym6a
         XL3B4LDJWCqA+ER6zG+pByocjOx0S9BHMfzxEU6cO0qATqEvsnzkTvCqOEBwxQUICRBU
         QIz+G5nTY791j0PrNcJUbfh8vUPQeRABPlkeim9RRyLow2xsUfYTrPeFI7QlZbCbgxgr
         t+HPS3x5o4NZA8IknvLX94eLHXvekcDfUw9V9N/ZdjkeUuk8Fa+10GQiuEwovU27ur3Z
         YstVFyMi7t37mYfcWNfy1dCYy3uLduyQ4tovjwdzkUfK6qopuLroLhz1eGYuUPlzqMvv
         aHBA==
X-Google-DKIM-Signature: v=1; a=rsa-sha256; c=relaxed/relaxed;
        d=1e100.net; s=20161025;
        h=x-gm-message-state:from:to:cc:subject:date:message-id;
        bh=4xx/AtucN5CvwisH9ffjv+eJjFMQ1dh59OgKza6nQas=;
        b=DwSyK187l1r9kG9IEsLoG4AkM06KxjjELYtnx9a335Ol61RVeVaJh7bYkWvv/cBoWS
         RphFTgSDEfNMiStO3iw4Y19z97xWyHjd+Z39Jllmyg95XEVjmk95MxgcnBKYSe+r+j+r
         aQ01KwTvqiPb6Tb/H5TaBTjjuX5eaFcS6Kl3QTXEHnB0NOa9CIgYmN121t/3DnvXDGcO
         7lr1vE4tYNkrh/EwZ4foO0UTvSA0WLw62wb4+wQaRkP2Xr3wGw90cjy9fEOEzZc7CPZ5
         KrHYdsa/gn3avXphj8aoelVzZ0GqdrQ0HThPUeW32/oVYCscinTR3/pGwei2GpCEIP/0
         MvKQ==
X-Gm-Message-State: APzg51DuFmFpFsE+vXAqvAM+gJ4swMg9buS86Avm91AEpJ44hioN7KRA
        leX+rFzIXAdfuTkRfMlcdw40VGGH4bQ=
X-Google-Smtp-Source: 
 ANB0VdavJ8W5gYCWyrabbBPeKbdVU58U4irmvPHc4ewatMWSJOosubETlS4mB6KZjM7Eoa1mBswokQ==
X-Received: by 2002:a65:4384:: with SMTP id
 m4-v6mr11803730pgp.265.1535981758112;
        Mon, 03 Sep 2018 06:35:58 -0700 (PDT)
Received: from localhost (65.49.233.5.16clouds.com. [65.49.233.5])
        by smtp.gmail.com with ESMTPSA id
 d9-v6sm47279087pfb.86.2018.09.03.06.35.56
        (version=TLS1_2 cipher=ECDHE-RSA-AES128-GCM-SHA256 bits=128/128);
        Mon, 03 Sep 2018 06:35:57 -0700 (PDT)
From: Wei Yang <richard.weiyang@gmail.com>
To: pbonzini@redhat.com, rkrcmar@redhat.com
Cc: tglx@linutronix.de, mingo@redhat.com, x86@kernel.org,
        kvm@vger.kernel.org, Wei Yang <richard.weiyang@gmail.com>
Subject: [PATCH] KVM: x86: don't reset root in kvm_mmu_setup()
Date: Mon,  3 Sep 2018 21:35:49 +0800
Message-Id: <20180903133549.23966-1-richard.weiyang@gmail.com>
X-Mailer: git-send-email 2.15.1
Sender: kvm-owner@vger.kernel.org
Precedence: bulk
List-ID: <kvm.vger.kernel.org>
X-Mailing-List: kvm@vger.kernel.org
X-Virus-Scanned: ClamAV using ClamSMTP

Here is the code path which shows kvm_mmu_setup() is invoked after
kvm_mmu_create(). Since kvm_mmu_setup() is only invoked in this code path,
this means the root_hpa and prev_roots are setup properly. And it is not
necessary to reset it again.

    kvm_vm_ioctl_create_vcpu()
        kvm_arch_vcpu_create()
            vmx_create_vcpu()
                kvm_vcpu_init()
                    kvm_arch_vcpu_init()
                        kvm_mmu_create()
        kvm_arch_vcpu_setup()
            kvm_mmu_setup()
                kvm_init_mmu()

This patch set reset_roots to false in kmv_mmu_setup().

Signed-off-by: Wei Yang <richard.weiyang@gmail.com>
---
 arch/x86/kvm/mmu.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/arch/x86/kvm/mmu.c b/arch/x86/kvm/mmu.c
index a282321329b5..28da8a9bd37b 100644
--- a/arch/x86/kvm/mmu.c
+++ b/arch/x86/kvm/mmu.c
@@ -5413,7 +5413,7 @@ void kvm_mmu_setup(struct kvm_vcpu *vcpu)
 {
 	MMU_WARN_ON(VALID_PAGE(vcpu->arch.mmu.root_hpa));
 
-	kvm_init_mmu(vcpu, true);
+	kvm_init_mmu(vcpu, false);
 }
 
 static void kvm_mmu_invalidate_zap_pages_in_memslot(struct kvm *kvm,
